components/script/dom/history.rs

```diff
@@ -48,9 +61,61 @@ impl History {
         let _ = self.window.upcast::<GlobalScope>().script_to_constellation_chan().send(msg);
         Ok(())
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-history-pushstate
+    // https://html.spec.whatwg.org/multipage/#dom-history-replacestate
+    fn push_or_replace_state(&self,
+                             cx: *mut JSContext,
+                             data: HandleValue,
+                             _title: DOMString,
+                             _url: Option<USVString>,
+                             _push_or_replace: PushOrReplace) -> ErrorResult {
+        // Step 1
+        let document = self.window.Document();
+        // Step 2
+        if !document.is_fully_active() {
+            return Err(Error::Security);
+        }
+
+        // TODO: Step 3 Optionally abort these steps
```

Comment:
Submit issues for those `TODO`s and cite them next to the `TODO`?

Reply:
Will do!